sink_service/source/main.c

```diff
@@ -60,17 +61,75 @@ static bool get_service_name(char service_name[MAX_SIZE_SERVICE_NAME], unsigned
     return true;
 }
 
+
+/**
+ * \brief   Obtains environment parameters to control sink settings
+ * \param   bitrate
+ *          Pointer where to store bitrate value (if any)
+ * \param   port_name
+ *          Pointer where to store port_name value (if any)
+ * \param   sink_id
+ *          Pointer where to store sink_id value (if any)
+ * \param   max_poll_fail_duration
+ *          Pointer where to store max_poll_fail_duration value (if any)
+ * \return  Number of acquired parameters
+ */
+static bool get_env_parameters(unsigned long * bitrate, char * port_name, unsigned int * sink_id, unsigned int * max_poll_fail_duration)
```

Comment:
Would it be possible to make a more generic function?

something like:
get_env_parameter(parameter_name, var_to_load, max size)

It would avoid all those copy/paste but you may have the issue with strtoul depending if you want a string or uint value.

Reply:
That's a good idea :+1: 

What do you think about leaving the conversion up to the caller?

Reply:
Yes, that would be good. So always retrieve a string and do conversion case by case